x/reward/abci.go

```diff
@@ -76,10 +69,16 @@ func handleKeyMgmtInflation(ctx sdk.Context, k types.Rewarder, m types.Minter, s
 	mintParams := m.GetParams(ctx)
 	totalAmount := minter.BlockProvision(mintParams).Amount.ToDec().Mul(k.GetParams(ctx).KeyMgmtRelativeInflationRate)
 
+	isProxyActive := func(v snapshot.ValidatorI) bool {
+		proxy, isActive := ss.GetProxy(ctx, v.GetOperator())
+
+		return isActive && !mSig.HasOptedOut(ctx, proxy)
+	}
+
 	var validators []snapshot.ValidatorI
 
 	validatorIterFn := func(_ int64, validator stakingtypes.ValidatorI) bool {
-		if excludeJailedOrTombstoned(ctx, slasher, ss, validator) {
+		if excludeJailedOrTombstoned(ctx, slasher, validator) || !isProxyActive(validator) {
```

Comment:
```suggestion
		if excludeJailedOrTombstoned(ctx, slasher, validator) {
	                return false
	        }
	        
	        if !isParticipatingKeygen(validator) {
	                return false
	        }
```
It's clearer to do multiple if's then doing playing with logical expression